File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException("Strands must be the same length.");
        }

        return firstStrand.Zip(secondStrand).Count(pair => pair.First != pair.Second);
    }
}

File: python/hamming/hamming.py
def distance(strand_a, strand_b):
    """Calculate Hamming distance between DNA strands"""
    if len(strand_a) != len(strand_b):
        raise ValueError('Strands must be equal length')
    return sum(1 for a, b in zip(strand_a, strand_b) if a != b)
